File: HorseRacing.Web/src/interfaces/api/contracts/model/game/requests/create-game/create-game-request-dto.ts
import type { BaseDto } from "~/interfaces/api/contracts/base/dto/base-dto";

export interface CreateGameRequestDto extends BaseDto{
    /** Идентификатор пользователя создавшего игру */
    UserId: string;
    /** Название игры */
    Name: string;
    /** Ставка пользователя создавшего игру */
    BetAmount: number;
    /** Масть лошади, которую выбрал пользователь при создании игры */
    BetSuit: number;
}

File: HorseRacing.Contracts/Models/Game/Requests/CreateGame/CreateGameRequestDto.cs
using HorseRacing.Contracts.Base.Dto;
using HorseRacing.Domain.GameAggregate.Enums;

namespace HorseRacing.Contracts.Models.Game.Requests.CreateGame
{
    public class CreateGameRequestDto : BaseDto
    {
        /// <summary>
        /// Идентификатор пользователя создавшего игру
        /// </summary>
        public Guid UserId { get; set; }
        /// <summary>
        /// Название игры
        /// </summary>
        public string Name { get; set; } = string.Empty;
        /// <summary>
        /// Ставка пользователя создавшего игру
        /// </summary>
        public int BetAmount { get; set; } = 10;
        /// <summary>
        /// Режим игры
        /// <see cref="GameModeType"/>
        /// </summary>
        public int Mode { get; set; } = 0;
        /// <summary>
        /// Масть лошади, 
        /// которую пользователь выбрал при создании игры
        /// <see cref="SuitType"/>
        /// </summary>
        public int BetSuit { get; set; } = 0;
    }
}
